### PR TITLE
Added AHCCD functions

### PR DESCRIPTION
**Pull request by KevinShook**
## Description

I've added 4 functions to read in AHCCD daily and monthly precips and temps.
The data are in EC's awful month x day or year x month formats.
The header may be in English or English and French

## Example

Each function reads in a single .txt file, e.g.
monthly_total_precip <- read_AHCCD_monthly_temp("mt3011120.txt")

Unfortunately we can't distribute the daily values, and the monthly files are contained
in a single huge .zip file, so I can't think of how to incorporate automated tests. I'm attaching some test files, but it might be a good idea to get some other ones, esp. as they change the file formats without warning. The data location is indicated in the documentation for each function.

The first character of the file name indicates its time step (d - daily, m- monthly).
The second indicates the data type, whether precip or temp and the type of value, either
min, max or mean.
[mt3011120.txt](https://github.com/CSHS-CWRA/CSHShydRology/files/3081090/mt3011120.txt)
[ms3011120.txt](https://github.com/CSHS-CWRA/CSHShydRology/files/3081091/ms3011120.txt)
[dt1126150.txt](https://github.com/CSHS-CWRA/CSHShydRology/files/3081092/dt1126150.txt)
[dm1085835.txt](https://github.com/CSHS-CWRA/CSHShydRology/files/3081093/dm1085835.txt)

### CODE REVIEW

File: R/read_AHCCD_daily_precip.R [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+#'Atmosphere-Ocean, 49 (2), 163-177.}
+#' @examples
+#' \dontrun{
+#'stoon_rainfall <- read_AHCCD_daily_precip("dr_4057120.txt")}
+#' @export
+
+read_AHCCD_daily_precip <- function(precip_file){
+  # check parameter
+  if (precip_file == "" | is.null(precip_file)){
+    cat('Error: precip_file missing\n')
```
Comment (boshek) [inline R/read_AHCCD_daily_precip.R]: I think you want `stop` here. That's what I would use.

File: R/read_AHCCD_daily_precip.R [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  # set up widths to read
+  header <- c(yearWidth,3,1)
+  header_classes <- c('numeric','numeric','character')
+  
+  cols <- rep.int(c(8,1),31)
+  cols_classes <- rep.int(c('numeric', 'character'), 31)
+  all <- c(header,cols)
+  all_classes <- c(header_classes, cols_classes)
+  
+  # read data
```
Comment (boshek) [inline R/read_AHCCD_daily_precip.R]: One other option here is to replace much of this code with a combination of `tidyr` `gather` and `spread`

File: R/read_AHCCD_daily_temp.R [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+#' Daily values must be requested. Any use of the temperature data must cite 
+#' \cite{Vincent, L. A., X. L. Wang, E. J. Milewska, H. Wan, F. Yang, and V. Swail, 2012. 
+#' A second generation of homogenized Canadian monthly surface air temperature for climate 
+#' trend analysis, J. Geophys. Res., 117, D18110, doi:10.1029/2012JD017859.}
+#' @examples
+#' \dontrun{
+#'stoon_tmax <- read_AHCCD_daily_temp("dx40657120.txt")}
+#' @export
+
+read_AHCCD_daily_temp <- function(temp_file){
```
Comment (boshek) [inline R/read_AHCCD_daily_temp.R]: You might also consider writing some helper functions as there is some redundant code between each function.

File: R/read_AHCCD_daily_precip.R (hunk 1 above)
Comment (KevinShook) [reply R/read_AHCCD_daily_precip.R]: Good idea. Will do.

File: R/read_AHCCD_daily_temp.R (hunk 3 above)
Comment (KevinShook) [reply R/read_AHCCD_daily_temp.R]: Also a good idea.

Comment (KevinShook) [reply R/read_AHCCD_daily_temp.R]: Done!

**Comment by KevinShook** [comment]
I've re-written the functions so that there are single functions to read in monthly and daily values, respectively. I've added a function to d/l a single station from the EC site as well. This could be improved by removing the need to specify the province. However, that would require getting the station list, which seems to only be accessible through the  WMS server.

**Comment by boshek** [comment]
Looks good to me!

**Pull request merged**